packages/template-retail-react-app/app/pages/checkout/partials/shipping-multi-address.jsx

```diff
@@ -0,0 +1,404 @@
+                            defaultMessage: 'Quantity'
+                        })}
+                        : {variant.quantity}
+                    </Text>
+                </ListItem>
+            )}
+        </List>
+    )
+}
+
+MultiShippingItemAttributes.propTypes = {
+    variant: PropTypes.object.isRequired,
+    includeQuantity: PropTypes.bool
+}
+
+const ShippingMultiAddress = ({
+    basket,
+    onSubmit,
+    submitButtonLabel,
+    addNewAddressLabel,
+    noItemsInBasketMessage,
+    deliveryAddressLabel
+}) => {
+    const {formatMessage} = useIntl()
+    const {currency} = useCurrency()
+    const productIds = basket?.productItems?.map((item) => item.productId).join(',')
+    const {
+        data: productsMap,
+        isLoading: productsLoading,
+        error: productsError
+    } = useProducts(
+        {parameters: {ids: productIds, allImages: true}},
+        {
+            enabled: Boolean(productIds),
+            select: (data) => {
+                return (
+                    data?.data?.reduce((acc, p) => {
+                        acc[p.id] = p
+                        return acc
+                    }, {}) || {}
+                )
+            }
+        }
+    )
+    const {data: customer} = useCurrentCustomer()
+    const addresses = customer?.addresses || []
+    const [selectedAddresses, setSelectedAddresses] = useState({})
+
+    if (!basket?.productItems?.length) {
+        return (
+            <Center
+                p={8}
+                textAlign="center"
+                color="gray.500"
+                role="status"
+                aria-live="polite"
+                aria-label={formatMessage(noItemsInBasketMessage)}
+            >
+                <VStack spacing={4}>
+                    <Text fontSize="lg" fontWeight="medium">
+                        {formatMessage(noItemsInBasketMessage)}
+                    </Text>
+                </VStack>
+            </Center>
+        )
+    }
+
+    // Loading state
+    if (productsLoading) {
+        return (
+            <Center p={8} textAlign="center" color="gray.500">
+                <VStack spacing={4}>
+                    <Spinner size="lg" />
+                    <Text>
+                        {formatMessage({
+                            id: 'shipping_multi_address.loading.message',
+                            defaultMessage: 'Loading products...'
+                        })}
+                    </Text>
+                </VStack>
+            </Center>
+        )
+    }
+
+    // Error state
+    if (productsError) {
+        return (
+            <Alert
+                status="error"
+                variant="subtle"
+                flexDirection="column"
+                alignItems="center"
+                justifyContent="center"
+                textAlign="center"
+                height="200px"
+                aria-live="assertive"
+            >
+                <AlertIcon boxSize="40px" mr={0} />
```

Comment:
Can we use theme value here? boxSize={5} or {6}? 
```suggestion
                <AlertIcon boxSize="40px" mr={0} />
```